documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc

```diff
@@ -63,39 +63,38 @@ For Maven based projects add the following to your POM file in order to use MapS
 <build>
     <plugins>
         <plugin>
-  <groupId>org.bsc.maven</groupId>
-  <artifactId>maven-processor-plugin</artifactId>
-  <version>2.2.4</version>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <version>3.5.1</version>
             <configuration>
-                <defaultOutputDirectory>
-                    ${project.build.directory}/generated-sources
-                </defaultOutputDirectory>
-                <processors>
-                    <processor>org.mapstruct.ap.MappingProcessor</processor>
-                </processors>
+                <source>1.8</source>
+                <target>1.8</target>
+                <annotationProcessorPaths>
+                    <path>
+                        <groupId>org.mapstruct</groupId>
+                        <artifactId>mapstruct-processor</artifactId>
+                        <version>${org.mapstruct.version}</version>
+                    </path>
+                </annotationProcessorPaths>
             </configuration>
-            <executions>
-                <execution>
-                    <id>process</id>
-                    <phase>generate-sources</phase>
-                    <goals>
-                        <goal>process</goal>
-                    </goals>
-                </execution>
-            </executions>
-            <dependencies>
-                <dependency>
-                    <groupId>org.mapstruct</groupId>
-                    <artifactId>mapstruct-processor</artifactId>
-                    <version>${org.mapstruct.version}</version>
-                </dependency>
-            </dependencies>
         </plugin>
     </plugins>
 </build>
+...
 ----
 ====
 
+[TIP]
+====
+If you are working with the Eclipse IDE, make sure to have a current version of the http://www.eclipse.org/m2e/[M2E plug-in].
+When importing a Maven project configured as shown above, it will set up the MapStruct annotation processor so it runs right in the IDE, whenever you save a mapper type.
+Neat, isn't it?
```

Comment:
It's awesome, indeed! But this would only work if you'd configure the default in `Preferences -> Maven -> Annotation Processing` to "Automatically configure JDT APT".
But, if you add this to the pom properties, it'll choose that behaviour nontheless:
```xml
  <properties>
    <mapstruct.version>1.0.0.Final</mapstruct.version>

    <!-- Automatically activate m2e APT in Eclipse  -->
    <m2e.apt.activation>jdt_apt</m2e.apt.activation>
  </properties>
```